### PR TITLE
STRY0019433 - profile_dists: fix for numerical identifiers being converted to floats instead of strings

### PR DESCRIPTION
**Pull request by emarinier**
**Description**

As a pipeline developer, I would like profile_dists to be updated so that numerical identifiers are treated as strings instead of floats in process_profiles, so that arborator and other downstream software keeps identifiers as strings.

**Acceptance Criteria**

- [x] Update process_profiles so that numerical identifiers are treated as strings
- [x] Add test case for numerical identifiers with code running process_profiles
- [x] Test out updated profile_dists with the arborator Python package
- [x] All other tests should continue to pass
- [ ] Release update on GitHub
- [ ] Release update on pypi
- [ ] Release update on bioconda

**Notes**

Addresses https://github.com/phac-nml/profile_dists/issues/44 and https://github.com/phac-nml/profile_dists/issues/45.

### CODE REVIEW

File: profile_dists/utils.py [hunk 1]
```diff
@@ -122,7 +122,11 @@ def update_column_map(c1,c2, missing_allele=MISSING_ALLELE, missing_allele_dista
     :param c2: dict
     :return: dict
     '''
-    allele_id = max(list(c1.values()))+1
+    if len(c1) == 0:
+        allele_id = 1
+    else:
+        allele_id = max(list(c1.values()))+1
```
Comment (emarinier) [inline profile_dists/utils.py:128]: This is partly related to removing the "0":0 allele mapping.

File: profile_dists/utils.py [hunk 2]
```diff
@@ -136,13 +140,11 @@ def is_all_columns_int(column_dtypes):
...
+
     for col in column_dtypes:
-        if col in VALID_INT_TYPES:
-            continue
-        count_non_int+=1
-    if count_non_int > 0:
-        return False
+        if col not in VALID_INT_TYPES:
+            return False
+
```
Comment (emarinier) [inline profile_dists/utils.py:147]: Simplified this function.

File: profile_dists/utils.py [hunk 3]
```diff
@@ -207,9 +212,9 @@ def combine_header(h1,h2):
     return combined
 
 def create_col_map(columns):
-    column_mapping ={}
+    column_mapping = {}
     for col in columns:
-        column_mapping[col] = {"0":0}
+        column_mapping[col] = {}
```
Comment (emarinier) [inline profile_dists/utils.py:217]: These seemed unused (and didn't show up at all when called from arborator). I noticed that 0 is the default flag for missing alleles, but it's still passing tests that handle empty/missing values (`data_empty_row_*.tsv`).

File: profile_dists/utils.py [hunk 4]
```diff
@@ -270,9 +284,9 @@ def process_profile(profile_path,format="text",column_mapping={}, missing_allele
         method = guess_format(List(unique_col_values))
         converted_allele_codes = convert_allele_codes(unique_col_values, method)
         if not column in column_mapping:
-            column_mapping[column] = converted_allele_codes
-        else:
-            update_column_map(column_mapping[column], converted_allele_codes)
+            column_mapping[column] = {}
+
+        update_column_map(column_mapping[column], converted_allele_codes)
```
Comment (emarinier) [inline profile_dists/utils.py:289]: This addresses a bug where allele maps were different depending on whether they were generated/called from profile_dists or arborator. This change should harmonize those differences.

File: profile_dists/version.py [hunk 5]
```diff
@@ -1 +1 @@
-__version__ = '1.0.8'
+__version__ = '1.0.10'
```
Comment (emarinier) [inline profile_dists/version.py:1]: The next version is 1.0.10.

File: tests/data/alphanumeric/results.text [hunk 6]
```diff
@@ -1,12 +0,0 @@
...
-B	0.0	0.0	0.0	0.0	0.0	25.0	25.0	25.0	25.0	25.0	25.0
-C	0.0	0.0	0.0	0.0	0.0	25.0	25.0	25.0	25.0	25.0	25.0
-D	0.0	0.0	0.0	0.0	0.0	25.0	25.0	25.0	25.0	25.0	25.0
-E	0.0	0.0	0.0	0.0	0.0	25.0	25.0	25.0	25.0	25.0	25.0
-1	25.0	25.0	25.0	25.0	25.0	0.0	0.0	0.0	0.0	0.0	0.0
-2	25.0	25.0	25.0	25.0	25.0	0.0	0.0	0.0	0.0	0.0	0.0
-3	25.0	25.0	25.0	25.0	25.0	0.0	0.0	0.0	0.0	0.0	0.0
-4	25.0	25.0	25.0	25.0	25.0	0.0	0.0	0.0	0.0	0.0	0.0
-5	25.0	25.0	25.0	25.0	25.0	0.0	0.0	0.0	0.0	0.0	0.0
-6.5	25.0	25.0	25.0	25.0	25.0	0.0	0.0	0.0	0.0	0.0	0.0
```
Comment (emarinier) [inline tests/data/alphanumeric/results.text:12]: These files were not used in pytest workflow (and were wrong).

File: tests/test_utils.py [hunk 7]
```diff
@@ -288,11 +288,11 @@ def test_if_file_ok():
...
-    (["1", "2", "3"], [1, 2, 3], 1, {"1": 1}, True),
-    (["1", "2", "3"], [1, 2, 3], 1, {"1": 1, "2": 2, "3": 3}, False),
-    (["1", "2", "3"], [4, 5, 6], 1, {}, True),
-    (["1", "2", "3"], [1, 2, 3], 0, {"1": 1, "2": 2, "3": 3}, False),
-    (["1", "2", "3"], [1, 2, 3], 1.1, {"1": 1}, True),
+    (np.array(["1", "2", "3"]), np.array([1, 2, 3]), 1, {"1": 1}, True),
+    (np.array(["1", "2", "3"]), np.array([1, 2, 3]), 1, {"1": 1, "2": 2, "3": 3}, False),
+    (np.array(["1", "2", "3"]), np.array([4, 5, 6]), 1, {}, True),
+    (np.array(["1", "2", "3"]), np.array([1, 2, 3]), 0, {"1": 1, "2": 2, "3": 3}, False),
+    (np.array(["1", "2", "3"]), np.array([1, 2, 3]), 1.1, {"1": 1}, True),
```
Comment (emarinier) [inline tests/test_utils.py:295]: This is fixing a warning where reflective lists were being passed into a function that wasn't suppose to use them.

File: tests/test_utils.py [hunk 8]
```diff
@@ -372,4 +372,4 @@ def test_filter_samples(labels, profiles, labels_remove, expected_labels, expect
     labels, profiles = utils.filter_samples(labels, profiles, labels_remove)
     assert expected_labels == labels
     for k, v in  zip(expected_profiles, profiles):
-        assert np.alltrue(k == v)
\ No newline at end of file
+        assert np.all(k == v)
```
Comment (emarinier) [inline tests/test_utils.py:375]: Fixing a deprecation warning.

**Review by apetkau — COMMENTED** [review]
This looks great. Thanks so much @emarinier 😄 

I have no in-line comments but do have a general comment. Since the changes touches some parts related to missing alleles (which should be encoded as `0`) could you make sure there are test cases that check the distances both using `--count_missing` (which counts missing alleles as a difference) and where `--count_missing` is unset/false (which won't count missing alleles as differences)? That way we can be sure this all works.

**Comment by emarinier** [comment]
> This looks great. Thanks so much @emarinier 😄
> 
> I have no in-line comments but do have a general comment. Since the changes touches some parts related to missing alleles (which should be encoded as `0`) could you make sure there are test cases that check the distances both using `--count_missing` (which counts missing alleles as a difference) and where `--count_missing` is unset/false (which won't count missing alleles as differences)? That way we can be sure this all works.

Added in ba9df47

**Review by apetkau — APPROVED** [review]
Thanks so much @emarinier . Looks great 😄
